php/libraries/Utility.class.inc

```diff
@@ -154,6 +155,39 @@ class Utility
         return $list;
     }
 
+    /**
+     * Returns a list of sites by site IDs from the database
+     *
+     * @param array $siteIDs a list of the study center IDs
+     *
+     * @return array an associative array("center ID" => "site name")
+     */
+    static function getSiteNameListByIDs(array $siteIDs): array
+    {
+        $factory = NDB_Factory::singleton();
+        $DB      = $factory->database();
+
+        $params = array();
+        $values = array();
+        foreach ($siteIDs as $key => $siteID) {
+            $para          = "cid$key";
+            $params[]      = ":$para";
+            $values[$para] = $siteID;
+        }
+
+        $sites = $DB->pselect(
+            "SELECT CenterID as ID, Name FROM psc WHERE CenterID in (" .
+            implode(', ', $params) . ")",
+            $values
+        );
+
+        $list = array();
+        foreach ($sites as $site) {
+            $list[$site['ID']] = $site['Name'];
+        }
```

Comment:
this should use pselectColWithIndexKey

https://github.com/aces/Loris/blob/master/php/libraries/Database.class.inc#L907

Reply:
I just tried to use `pselectColWithIndexKey`, but the installer doesn't work now, I have deleted my old DB because the links did not work. I don't see any difference between `pselect` and `pselectColWithIndexKey` for this particular case, so I just leave it for now.

Reply:
You can modify if you wish, I just don't see any necessity for this modify. The master branch still doesn't work after I fixed the installer, I am not able to test

Reply:
pselectColWIthIndexKey will return the array properly indexed in one step making lines 184 to 187 unnecessary.

I can make the changes it's fine and thanks for reporting the errors with the installer